src/main/java/PersonalBudget/business/expense/dto/ExpenseDTO.java

```diff
@@ -0,0 +1,16 @@
+package PersonalBudget.business.expense.dto;
+
+import jakarta.validation.constraints.NotEmpty;
+
+public record ExpenseDTO(
+
+        @NotEmpty(message = "{amount.empty}")
+        String amount,
+        @NotEmpty(message = "{date.empty}")
+        String expenseDate,
```

Comment:
This can be a LocalDate or LocalDateTime https://www.baeldung.com/spring-date-parameters
Spring should be able to map string sent from the application to a date, and if it's not it's a matter of configuration :)